src/v/utils/vint.cc

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "utils/vint.h"
+
+#include <seastar/core/coroutine.hh>
+
+namespace unsigned_vint {
+
+ss::future<std::pair<uint32_t, size_t>>
+stream_deserialize(ss::input_stream<char>& s) {
+    /// Consume up to 5 iterations (0-4) of reading 7 bits each time
+    constexpr auto limit = ((max_length - 1) * 7);
+    detail::var_decoder decoder(limit);
+    while (!s.eof()) {
+        auto buf = co_await s.read_exactly(1);
```

Comment:
can you double check that it isn't important to check that `buf` is zero length?

Reply:
I think its ok, in `seastar/core/iostream-impl.hh` looks like `_eof` is set to true when the internal buffer has a size of 0. So if i just checked it, i think i'm safe to assume the length of that buffer > 0

Reply:
Implemented the check